lib/ttfunk/table/maxp.rb

```diff
@@ -24,20 +24,118 @@ class Maxp < Table
       attr_reader :max_component_elements
       attr_reader :max_component_depth
 
-      def self.encode(maxp, mapping)
-        ''.b.tap do |table|
-          num_glyphs = mapping.length
-          table << [maxp.version, num_glyphs].pack('Nn')
-
-          if maxp.version == 0x10000
-            table << [
-              maxp.max_points, maxp.max_contours, maxp.max_component_points,
-              maxp.max_component_contours, maxp.max_zones,
-              maxp.max_twilight_points, maxp.max_storage,
-              maxp.max_function_defs, maxp.max_instruction_defs,
-              maxp.max_stack_elements, maxp.max_size_of_instructions,
-              maxp.max_component_elements, maxp.max_component_depth
-            ].pack('n*')
+      class << self
+        def encode(maxp, new2old_glyph)
+          ''.b.tap do |table|
+            num_glyphs = new2old_glyph.length
+            table << [maxp.version, num_glyphs].pack('Nn')
+
+            if maxp.version == 0x10000
+              stats = stats_for(
+                maxp, glyphs_from_ids(maxp, new2old_glyph.values)
+              )
+
+              table << [
+                stats[:max_points] || 0,
+                stats[:max_contours] || 0,
+                stats[:max_component_points] || 0,
+                stats[:max_component_contours] || 0,
```

Comment:
We have `value_or(0)` when we compose this hash. I guess we can drop these `|| 0` here.

Reply:
Unfortunately, for some fonts the `stats` hash won't contain some of these keys, so I had to add `|| 0`. I'll see if I can clean that up a bit.

Reply:
Hash with default value?

Reply:
Ok, just made some changes, let me know what you think :)